### ISSUE TITLE
Proposal: Error for ++ on Strings

### ISSUE THREAD

**Issue by aersamkull** [palantir/tslint#755]
Just a little proposal I first brought up here: https://github.com/Microsoft/TypeScript/issues/5374

I recently wrote code like this:

``` TypeScript
const aString = "asf" +
                + "asdf";
```

Which results in "asfNaN" which is not very obvious, I think.

I am even more confused as "1"+ +"2" is "12"!? However, that's JavaScript's fault, not TypeScript's. 

Is it possible to have a rule for that?
How do you think about it?

**Comment by DanielRosenwasser** [comment]
I could see a few different lint rules here:

1. Don't allow coercing a string literal to a number with `+` (just write a number)
2. Don't allow a unary expression to be on its own line.

**Comment by myitcv** [comment]
Related to #697

**Comment by adidahiya** [comment]
yeah it looks like #697 addresses the type coercion issue. I'll leave this issue open for discussion around related issues, like unary expressions on their own line

**Comment by tomijah** [comment]
```
import * as ts from 'typescript';
import * as Lint from 'tslint/lib/lint';

export class Rule extends Lint.Rules.AbstractRule {
  public static FAILURE_STRING = 'string unary plus operator not allowed';

  public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
    const walker = new NoStringunaryOperatorsWalker(sourceFile, this.getOptions());
    return this.applyWithWalker(walker);
  }
}

class NoStringunaryOperatorsWalker extends Lint.RuleWalker {
  protected visitPrefixUnaryExpression(node: ts.PrefixUnaryExpression) {
    const sourceFile = this.getSourceFile();
    const position = node.getStart(sourceFile);
    if (node.operator === ts.SyntaxKind.PlusToken && node.operand.kind === ts.SyntaxKind.StringLiteral) {
      this.addFailure(this.createFailure(Math.min(position, this.getLimit()), 0, Rule.FAILURE_STRING));
    }
  }
}
```

**Comment by JoshuaKGoldberg** [comment]
Looks like this hasn't gotten discussion in a while, and there are other discussions around styles there. 💪
